nilearn/regions/tests/test_region_extractor.py

```diff
@@ -0,0 +1,115 @@
+from nilearn.regions.region_extractor import _threshold_maps
+from nilearn.image import iter_img
+
+from nilearn._utils.testing import assert_raises_regex, generate_maps
+
+
+def _make_random_data(shape):
+    affine = np.eye(4)
+    rng = np.random.RandomState(0)
+    data_rng = rng.normal(size=shape)
+    img = nibabel.Nifti1Image(data_rng, affine)
+    data = img.get_data()
+    return img, data
+
+
+def test_invalid_thresholds_in_threshold_maps():
+    shape = (91, 109, 91)
+    maps = generate_maps(shape, n_regions=2)
+    map_0 = maps[0]
+
+    for invalid_threshold in [10, '80%', 'auto']:
+        assert_raises_regex(ValueError,
+                            "Threshold must be float value",
+                            _threshold_maps,
+                            map_0, threshold=invalid_threshold)
+
+
+def test_passing_threshold_maps():
+    # smoke test for function _threshold_maps with randomly
+    # generated maps
+    shape = (6, 8, 10)
+    maps = generate_maps(shape, n_regions=3)
+    map_0 = maps[0]
+    for map_img, threshold in zip(iter_img(map_0), [0.0, 0.5, 1.0]):
+        thr_maps = _threshold_maps(map_img, threshold=threshold)
+
+
+def test_invalids_extract_types_in_connected_regions():
+    shape = (91, 109, 91)
+    n_regions = 2
+    maps = generate_maps(shape, n_regions)
+    map_0 = maps[0]
+    valid_names = ['connected_components', 'local_regions']
+
+    # test whether same error raises as expected when given invalid inputs to
+    # extract_type of regions extraction
+    message = ("'extract_type' should be {0}").format(valid_names)
+    for img in iter_img(map_0):
+        assert_raises_regex(ValueError,
+                            message,
+                            connected_regions,
+                            img, min_size=40,
+                            extract_type='connect_regions')
+
+    # test whether error raises when there is an invalid inputs to extract
+    invalid_extract_type = ['local', 'asdf', 10]
+    for type_ in invalid_extract_type:
+        assert_raises(ValueError, connected_regions,
+                      img, extract_type=type_)
+
+
+def test_passing_connected_regions():
+    shape = (91, 109, 91)
+    n_regions = 4
+    maps = generate_maps(shape, n_regions)
+    map_0 = maps[0]
+
+    valid_extract_types = ['connected_components', 'local_regions']
+    # smoke test for function connected_regions
+    for map_, type_ in zip(iter_img(map_0), valid_extract_types):
+        regions_extracted = connected_regions(
+            map_, min_size=30, extract_type=type_, peak_local_smooth=3)
+
+
+def test_passing_RegionExtractor_object():
+    shape = (91, 109, 91)
+    n_regions = 2
+    n_subjects = 5
+    maps = generate_maps(shape, n_regions)
+    map_0 = maps[0]
+
+    affine = np.eye(4)
+    mask_img = nibabel.Nifti1Image(np.ones((shape), dtype=np.int8), affine)
+
+    # smoke test to RegionExtractor with thresholding_strategy='ratio_n_voxels'
```

Comment:
Idem as the previous test, hopefully we can do better than a smoke test for RegionExtractor.